tests/SdAiAgent/Abilities/WpRestAbilitiesTest.php

```
@@ -0,0 +1,495 @@
+	 * Test: loop guard rejects a call when AgentController is on the stack.
+	 *
+	 * This simulates the AgentController frame by temporarily adding a filter
+	 * that invokes WpRestAbilities::handle_execute() from inside a mock
+	 * AgentController-named context via reflection.
+	 */
+	public function test_loop_guard_rejects_recursive_call(): void {
+		// Use a closure that appears in debug_backtrace() as coming from AgentController.
+		$result = null;
+
+		// Invoke handle_execute from within a fake AgentController frame.
+		$fake_controller = new class() {
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$fake_controller'. (undefined)

(UnusedLocalVariable)